core/clustersmngr/clustersmngr.go

```diff
@@ -72,14 +74,19 @@ type clientsPool struct {
 	mutex   sync.Mutex
 }
 
-type ClusterClientConfig func(Cluster) *rest.Config
+type ClusterClientConfig func(Cluster) (*rest.Config, error)
```

Comment:
I spent time trying to understand this change and finally realized that this type is a function.

Could you use this opportunity to rename this type to `ClusterClientConfigFunc`, please?
In Golang we have similar conventions like context's `CancelFunc` and http's `HandlerFunc`, for example.